src/ResultSet.php

```diff
@@ -48,4 +48,27 @@ public function getIterator(): ArrayIterator
     {
         return new ArrayIterator($this->results);
     }
+
+    public function getErrors(): self
+    {
+        $resultSet = new self();
+        foreach ($this->results as $attribute => $result) {
+            if (!$result->isValid()) {
+                $resultSet->addResult($attribute, $result);
+            }
+        }
```

Comment:
Are you sure that you want to get ResultSet instead of an array?

Reply:
It seems logical to me

Reply:
Why not just flatten array? It will be already invalid so you need only attributes and their results.

Reply:
fixed